support/cas-server-support-pm-webflow/src/main/java/org/apereo/cas/pm/web/flow/PasswordManagementSingleSignOnParticipationStrategy.java

```diff
@@ -0,0 +1,53 @@
+package org.apereo.cas.pm.web.flow;
+
+import org.apereo.cas.CentralAuthenticationService;
+import org.apereo.cas.ticket.TransientSessionTicket;
+import org.apereo.cas.web.flow.SingleSignOnParticipationStrategy;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.webflow.execution.RequestContext;
+
+/**
+ * This is {@link PasswordManagementSingleSignOnParticipationStrategy}.
+ *
+ * @author Julien Huon
+ * @since 6.3.0
+ */
+@Slf4j
+@RequiredArgsConstructor
+public class PasswordManagementSingleSignOnParticipationStrategy implements SingleSignOnParticipationStrategy {
+    private final CentralAuthenticationService centralAuthenticationService;
+
+    @Override
+    public boolean supports(final RequestContext requestContext) {
+        LOGGER.trace("Evaluating if the Password Reset request is valid");
+        val transientTicket = requestContext
+            .getRequestParameters()
+            .get(PasswordManagementWebflowUtils.REQUEST_PARAMETER_NAME_PASSWORD_RESET_TOKEN);
+
+        if (StringUtils.isBlank(transientTicket)) {
+            LOGGER.trace("Password reset token is missing");
+            return false;
+        }
+
+        try {
+            val ticket = centralAuthenticationService.getTicket(transientTicket, TransientSessionTicket.class);
+            if (ticket != null || !ticket.isExpired()) {
+                LOGGER.trace("Token ticket [{}] is valid, SSO will be disabled", transientTicket);
+                return true;
+            }
```

Comment:
These lines are unnecessary. The ticket that is returned from `CAS` should always be valid, or else exceptions are thrown.

Reply:
Even with a JPA Ticket Registry where an expired token is not immediately deleted but deleted periodically with a cleaner?

But I could do it, sure.



There is only one thing i'm not confortable with. Aren't we doing the same verification 3 times?

1. At the right beginning of the flow with the ValidatePasswordResetTokenAction
2. Inside the Single Sign On Participation Stategy 
3. In the VerifyPasswordResetRequestAction wich finaly put in the flow with `PasswordManagementWebflowUtils.putPasswordResetToken(requestContext, token)` and then it's used by the InitPasswordResetAction with `PasswordManagementWebflowUtils.getPasswordResetToken(requestContext, token)`.

Isn't it a better design if the Validate Password Reset Token Action put the token in the flow if it is valid and then the Single Sign On Participation Strategy & the verification password reset request action get it from the flow with the getPasswordResetToken method?

Reply:
> Even with a JPA Ticket Registry where an expired token is not immediately deleted but deleted periodically with a cleaner?

Yes, could be. Not sure how this is related to this section?

> Isn't it a better design if the Validate Password Reset Token Action put the token in the flow if it is valid and then the Single Sign On Participation Strategy & the verification password reset request action get it from the flow with the getPasswordResetToken method?

Seems like it, yes. You are welcome to rework this all in a separate pull request.

Reply:
Never used the JPA Ticket Registry but the doc explain there is a cleaner which periodically delete the expired token. 

Which mean to me that an expired token could exist in the Ticket Registry if it has not been cleaned yet.

Also, the getTicket method documentation tel us:

```
/**
     * Obtains the given ticket by its id
     * and returns the CAS-representative object. **Implementations
     * need to check for the validity of the ticket by making sure
     * it exists and has not expired yet**, etc. This method is specifically
     * designed to remove the need to access the ticket registry.
**/
```

Reply:
Still not sure how it's related. You should examine the source code for the CAS implementation component to see how it basically does the same checks. If you do use CAS, the checks you have here (and should be removed) are handled automatically for you. It has nothing to do with JPA, etc.

Reply:
The source code of CAS is full of these kind of check:

Reply:
- You're checking if the ticket is not null and not expired after you get a ticket.
- If you do centralAuthenticationService.getTicket(), those checks are automatically done for you. You should not have to repeat them.

Reply:
> The source code of CAS is full of these kind of check. For instance :

All such checks should be cleaned up too; they are duplicates.

Reply:
I can handle the clean up bit to remove these checks. I think the reason they exist is because at some point, references to ticketRegistry were swapped with centralAuthenticationService to handle transactions better when fetching tickets. Seems like the duplicate checks were not cleaned up, so we should remove those (which would also improve coverage as well). 

Thanks for the catch! Looking good so far.

Reply:
Ok, the check is done in the verifyTicketState method of the AbstractCentralAuthenticationService class. :)

I guess a code cleanup should be done in oauth, pac4j etc...

Reply:
Right now I'm working on a better support of oauth/oidc because it would be nice to see CAS in the certified OIDC Provider lists.

Later I'll try the design improvement we discussed here.

Reply:
Done